src/stan-users-guide/style-guide.Rmd

```diff
@@ -214,33 +218,34 @@ transformed parameters {
 }
 ```
 
+The exception to this rule is local blocks which only exist for
+scoping reasons. The opening brace of these blocks is not associated
+with any control flow or block structure, so it should appear on its 
+own line.
 
 
 ## Conditionals
 
-Stan supports the full C++-style conditional syntax,
-allowing real or integer values to act as conditions, as follows.
+While Stan supports the full C++-style conditional syntax,
+allowing real or integer values to act as conditions, this
+should be avoided.
+
+The condtional expression should never be a value that is not produced
+ as the result of a comparison or boolean operation. Therefore, use
```

Comment:
Seems to be an extra line-initial space here.  I don't like the general guidance, though.  I think we should allow something like this:

```stan
if (bernoulli_rng(0.5)) {
   // do something
}
```

I think it's also totally OK to have this:

```stan
data {
  array[N]  int<lower = 0, upper = 1> foo;
  ...
model {
  for (n in 1:N) {
    if (foo[n]) {
      // do something
    }
  }
}
```

I think all we need to say is that rather than a real expression `e`, we should have `e != 0`.